syntax = "proto3";

import "common_messages.proto";

package MapReduce;

service MasterServices {
    rpc SubmitMapJob(SubmitMapJobArgs) returns (SubmitMapReply);
    rpc SubmitReduceJob(SubmitReduceJobArgs) returns (SubmitReduceReply);
}

message SubmitMapJobArgs {
    int32 worker_id = 1;
}

message SubmitMapReply {
}

message SubmitReduceJobArgs {
    int32 worker_id = 1;
}

message SubmitReduceReply {
}